### ISSUE TITLE
Damage application needs improvements

### ISSUE THREAD

**Issue by InconspicuousBear** [foundryvtt/dnd5e#3966]
Hello, it would be great to have the ability to select which damage part of a spell to apply instead of having to do it manually.

Here is an example:

Hunger of Hadar. It has 2 Damage instancers which do not always apply. The system allows just one damage roll.

Which rolls both instances at the same time. And then if you click apply, it applys everything. Even though half of the rolls may not apply.
![brave_o9fpUNtu2z](https://github.com/user-attachments/assets/1691472f-dbc3-4dba-b778-9a4e154c0f71)

**Comment by SlamHammerfist** [comment]
Assuming you mean that you have resistance or immunity to one or more of these types when you said, "even though half of the rolls may not apply." I just tested this out and it appears to be working as intended. I casted hunger of hadar on a character with acid resistance and applied damage. they took the full cold damage and half of the acid damage. Tested it with cold immunity and they only took the acid damage. Perhaps you have a mod that's affecting this, try without any mods on. Or am I misunderstanding the issue?

**Comment by InconspicuousBear** [comment]
No, I mean the fact that it rolls one big number. But a creature can walk out and then only the cold damage would apply. That is the issue.

**Comment by SlamHammerfist** [comment]
Oh, I see, they take damage at different times. I think this might be fixed with the new activities feature coming in 4.0.

**Comment by Fyorl** [comment]
Conditional damage should be moved into a separate activity.
